crypto/evp_extra/p_kem_test.cc

```diff
@@ -0,0 +1,502 @@
+  ASSERT_EQ(CBS_len(&expanded_key), test.secret_key_len);
+
+  // Verify it matches the original secret key
+  EXPECT_EQ(Bytes(CBS_data(&expanded_key), CBS_len(&expanded_key)),
+            Bytes(pkey->pkey.kem_key->secret_key, test.secret_key_len));
+}
+
+TEST_P(KEMTest, ParsePublicKey) {
+  // Test parsing of the draft standard example public keys
+
+  const KEMTestVector &test = GetParam();
+  int nid = test.nid;
+  size_t public_key_len = test.public_key_len;
+  size_t secret_key_len = test.secret_key_len;
+  const char *public_pem_str = test.public_pem_str;
+
+  // ---- 1. Convert example PEM to DER ----
+  uint8_t *der = nullptr;
+  long der_len = 0;
+  ASSERT_TRUE(PEM_to_DER(public_pem_str, &der, &der_len));
+  bssl::UniquePtr<uint8_t> free_der(der);
+
+  // ---- 2. Parse the public key ----
+  CBS cbs;
+  CBS_init(&cbs, der, der_len);
+  bssl::UniquePtr<EVP_PKEY> pkey_from_der(EVP_parse_public_key(&cbs));
+  ASSERT_TRUE(pkey_from_der);
+  ASSERT_EQ(EVP_PKEY_id(pkey_from_der.get()), EVP_PKEY_KEM);
+
+  // ---- 3. Verify key parameters ----
+  KEM_KEY *kem_key = pkey_from_der->pkey.kem_key;
+  ASSERT_TRUE(kem_key);
+  ASSERT_EQ(kem_key->kem->nid, nid);
+  ASSERT_EQ(kem_key->kem->public_key_len, public_key_len);
+  ASSERT_EQ(kem_key->kem->secret_key_len, secret_key_len);
+}
+
+TEST_P(KEMTest, ParseExamplePrivateKey) {
+  // Test parsing of the draft standard example private keys (expanded format)
+
+  const KEMTestVector &test = GetParam();
+  int nid = test.nid;
+  size_t public_key_len = test.public_key_len;
+  size_t secret_key_len = test.secret_key_len;
+  const char *private_pem_expanded_str = test.private_pem_expanded_str;
+
+  // ---- 1. Convert example PEM to DER ----
+  uint8_t *der = nullptr;
+  long der_len = 0;
+  ASSERT_TRUE(PEM_to_DER(private_pem_expanded_str, &der, &der_len));
+  bssl::UniquePtr<uint8_t> free_der(der);
+
+  // ---- 2. Parse the private key ----
+  CBS cbs;
+  CBS_init(&cbs, der, der_len);
+  bssl::UniquePtr<EVP_PKEY> pkey_from_der(EVP_parse_private_key(&cbs));
+  ASSERT_TRUE(pkey_from_der);
+  ASSERT_EQ(EVP_PKEY_id(pkey_from_der.get()), EVP_PKEY_KEM);
+
+  // ---- 3. Verify key parameters ----
+  KEM_KEY *kem_key = pkey_from_der->pkey.kem_key;
+  ASSERT_TRUE(kem_key);
+  ASSERT_EQ(kem_key->kem->nid, nid);
+  ASSERT_EQ(kem_key->kem->public_key_len, public_key_len);
+  ASSERT_EQ(kem_key->kem->secret_key_len, secret_key_len);
+
+  // ---- 4. Verify private key is present ----
+  ASSERT_NE(kem_key->secret_key, nullptr);
+}
+
+// Invalid length test vectors - truncated DER structures
+static const uint8_t mlkem512_public_key_invalid_length[] = {
+    0x30, 0x11, 0x30, 0x0b, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01,
+    0x65, 0x03, 0x04, 0x04, 0x01, 0x03, 0x02, 0x00, 0xe4};
+
+static const uint8_t mlkem512_private_key_invalid_length[] = {
+    0x30, 0x16, 0x02, 0x01, 0x00, 0x30, 0x0b, 0x06, 0x09, 0x60, 0x86, 0x48,
+    0x01, 0x65, 0x03, 0x04, 0x04, 0x01, 0x04, 0x04, 0x82, 0x45, 0x52, 0xd8};
+
```

Comment:
how were these generated? The additions to the fuzzing should provide coverage for test vectors like this, and we might be able to see if the length checks of the public/private parsing is triggered by those new additions to the corpus in this PR.